### PR TITLE
feat(API): Adding the flag to handle basic CORS policies in the framework

### PR DESCRIPTION
**Pull request by minrock**
Added a new flag to `falcon.API` class init method in order to turn on/off feature to handle basic CORS policies in the framework. By default, it will come off so all request must meet CORS policies.

Addresses: #1194

### CODE REVIEW

File: falcon/api.py [hunk 1]
```diff
@@ -136,6 +136,9 @@ def process_response(self, req, resp, resource, req_succeeded)
             when that same component's ``process_request()`` (or that of
             a component higher up in the stack) raises an exception.
 
+        cors_enable (bool): Set this flags to ``True`` if you want to accept
```
[the remark this answers is not in the document]
Comment (minrock) [reply falcon/api.py]: Fixed, already upload it on this PR
```python
        cors_enable (bool): Set this flag to ``True`` if you want to accept
            all requests with CORS policies.
```

File: tests/test_headers.py [hunk 2]
```diff
@@ -692,6 +704,28 @@ def test_content_length_options(self, client):
...
+
+    def test_enabled_cors_should_accept_all_origins_requests(self, cors_client):
+        cors_client.app.add_route('/', CORSHeaderResource())
+        result = cors_client.simulate_get()
+        assert result.headers['Access-Control-Allow-Origin'] == '*'
+
+    def test_enabled_cors_should_control_expiration_through_headers(self, cors_client):
+        cors_client.app.add_route('/', CORSHeaderResource())
+        result = cors_client.simulate_options(headers=(('Access-Control-Request-Method', '*'), ))
+        assert result.headers['Access-Control-Max-Age'] == '86400'  # 24 hours in minutes
```
Comment (vytas7) [inline tests/test_headers.py]: FWIW it is 24 hours in seconds :slightly_smiling_face:

Comment (minrock) [reply tests/test_headers.py]: You are right! Fixed it! thanks :)

File: docs/api/cors.rst [hunk 3]
```diff
@@ -0,0 +1,21 @@
+.. _cors:
+
+Cors
+=====
+
+Falcon allows an easy way to enable or disable your CORS policies. By
+default CORS policies are enable in the :any:`falcon.API` object so all request
+coming from other systems with different domains will be blocked. You can
```
[the remark this answers is not in the document]
Comment (minrock) [reply docs/api/cors.rst]: Changed in last push on the MR, please check if it's better now, thanks.

File: falcon/api.py [hunk 4]
```diff
@@ -136,6 +136,9 @@ def process_response(self, req, resp, resource, req_succeeded)
             when that same component's ``process_request()`` (or that of
             a component higher up in the stack) raises an exception.
 
+        cors_enable (bool): Set this flag to ``True`` if you want to accept
+            all requests with CORS policies.
```
[the remark this answers is not in the document]
Comment (minrock) [reply falcon/api.py]: Changed, let's check if it may work now. Thanks.

File: falcon/api.py [hunk 5]
```diff
@@ -288,6 +293,10 @@ def __call__(self, env, start_response):  # noqa: C901
 
                     req_succeeded = False
 
+            # If CORS are enabled then, we need to add the headers to the response
+            if self._cors_enable:
+                self._add_cors_headers(req, resp, req_succeeded)
```
[the remark this answers is not in the document]
Comment (minrock) [reply falcon/api.py]: Where do you think is a good place to create the middleware, should I create a new file name middlewares, so in that files we could create "commons" middlewares that may come in falcon out of the box?

[the remark this answers is not in the document]
Comment (minrock) [reply falcon/api.py]: Changed to a separated middleware and add it if the API call has cors_enabled is True

**Comment by kgriffs** [comment]
I went ahead and made a few modifications to clarify the docs and ensure full test coverage.

**Comment by kgriffs** [comment]
I'll add the news fragment in a followup PR after this is merged.

File: falcon/middlewares.py [hunk 6]
```diff
@@ -0,0 +1,29 @@
+
+class CORSMiddleware(object):
+    def process_response(self, req, resp, resource, req_succeeded):
+        """Implement a simple blanked CORS policy for all routes.
```
Comment (vytas7) [inline falcon/middlewares.py]: Typo _blanket_?

Comment (kgriffs) [reply falcon/middlewares.py]: heh, nice catch

**Pull request merged**